cfme/fixtures/vm.py

```diff
@@ -58,13 +67,29 @@ def _get_vm_name(request):
     return random_vm_name(req[0])
 
 
+# @pytest.fixture(scope="module")
+# def full_template_vm_modscope(setup_provider_modscope, request, full_template_modscope, provider):
+#     vm_name = _get_vm_name(request)
+#     return _create_vm(request, full_template_modscope, provider, vm_name)
+
+
+# @pytest.fixture(scope="function")
+# def full_template_vm(setup_provider, request, full_template, provider):
+#     vm_name = _get_vm_name(request)
+#     return _create_vm(request, full_template, provider, vm_name)
+
+
 @pytest.fixture(scope="module")
-def full_template_vm_modscope(setup_provider_modscope, request, full_template_modscope, provider):
+def create_vm_modscope(setup_provider_modscope, request, provider):
+    if request.param:
+        template_type = request.param
     vm_name = _get_vm_name(request)
-    return _create_vm(request, full_template_modscope, provider, vm_name)
+    return _create_vm(request, template_type, provider, vm_name)
 
 
 @pytest.fixture(scope="function")
-def full_template_vm(setup_provider, request, full_template, provider):
+def create_vm(setup_provider, request, provider):
+    if request.param:
+        template_type = request.param
     vm_name = _get_vm_name(request)
-    return _create_vm(request, full_template, provider, vm_name)
+    return _create_vm(request, template_type, provider, vm_name)
```

Comment:
Obviously need some defaulting behavior here, whether its test uncollect/skip, or trying to pick `'small_template'` from the yaml.

Reply:
skipping the test here makes sense, IMHO, cause if you are running a test with some template, you definitely need that template, defaulting to another one will not work. This is specially true for vm console tests I know, and may be applicable to others too.